mmv1/third_party/terraform/website/docs/guides/version_5_upgrade.html.markdown

```diff
@@ -242,6 +242,18 @@ all of the labels present on the resource in GCP including the labels configured
 through Terraform, the system, and other clients, equivalent to
 `effective_labels` on the resource.
 
+### TFC Drift Detection with Labels
+
+ Due to the provider not being authoritative in +5.0.0, it has led to issues in detecting drift with labels through Terraform Cloud's UI. In order to detect drift, it's recommended to use Terraform's `check` block to compare the number of labels in both `effective_labels` and `terraform_labels`. A mismatch would only happen if drift is present, the following code would run this check on applicable resources.
```

Comment:
This is a good start, though I think we need to flesh it out a little to help users understand _why_ drift detection (DD) wasn't 100% correct when used with `labels` fields in the past, explain the new recommended approach, and caveat everything with the idea that detecting drift on labels isn't necessarily helpful to them. Without that context I can understand a user being confused.

Some ideas of how to flesh out this section:
- I think it would be worth starting this section with a description of how DD used to be able to detect labels added out of band. Then describe how GCP would often add labels itself as part of its normal function, leading to warnings from DD that were not beneficial to users (i.e. it wasn't a sign of people manually editing resources, and if they removed the unexpected labels the systems in GCP would just re-add them). The purpose of that section would be to show how use of DD with labels wasn't 100% 'correct' in the past.
- Then I would move on to describing the workaround, with a caveat saying that it will re-introduce the problem described above.
- It's also worth describing what a user should do if a new label is detected via DD. They would need to determine if the change in labels is due to a person or due to GCP. If new labels added by GCP are stable then they should add it to their Terraform configuration to make the check pass again. In cases where GCP routinely changes the labels the user should consider if using DD on that resource is helping them or resulting in unnecessary noise.

Let me know what you think, or maybe we can pull in someone from the Google team to get more input here if that helps?